crates/oxc_linter/src/rules/oxc/only_used_in_recursion.rs

```diff
@@ -92,21 +96,54 @@ impl Rule for OnlyUsedInRecursion {
         }
 
         for (arg_index, formal_parameter) in function_parameters.items.iter().enumerate() {
-            let BindingPatternKind::BindingIdentifier(arg) = &formal_parameter.pattern.kind else {
-                continue;
-            };
+            match &formal_parameter.pattern.kind {
+                BindingPatternKind::BindingIdentifier(arg) => {
+                    if is_argument_only_used_in_recursion(function_id, arg, arg_index, ctx) {
+                        create_diagnostic(
+                            ctx,
+                            function_id,
+                            function_parameters,
+                            arg,
+                            arg_index,
+                            function_span,
+                        );
+                    }
+                }
+                BindingPatternKind::ObjectPattern(pattern) => {
+                    for property in &pattern.properties {
+                        let Some(ident) = get_binding_identifier(property) else {
+                            continue;
+                        };
+
+                        let Some(name) = property.key.name() else {
+                            continue;
+                        };
+                        if is_property_only_used_in_recursion_jsx(ident, &name, function_id, ctx) {
+                            create_diagnostic_jsx(ctx, function_id, property);
+                        }
+                    }
+                }
+                _ => continue,
+            }
+        }
+    }
+}
 
-            if is_argument_only_used_in_recursion(function_id, arg, arg_index, ctx) {
-                create_diagnostic(
-                    ctx,
-                    function_id,
-                    function_parameters,
-                    arg,
-                    arg_index,
-                    function_span,
-                );
+fn get_binding_identifier<'a>(
+    property: &'a BindingProperty<'a>,
+) -> Option<&'a BindingIdentifier<'a>> {
+    match &property.value.kind {
+        BindingPatternKind::BindingIdentifier(binding_identifier) => Some(binding_identifier),
+        BindingPatternKind::AssignmentPattern(assignment_pattern) => {
+            if let BindingPatternKind::BindingIdentifier(binding_identifier) =
+                &assignment_pattern.left.kind
+            {
+                Some(binding_identifier)
+            } else {
+                None
             }
         }
+        _ => None,
     }
 }
```

Comment:
does using https://github.com/oxc-project/oxc/blob/0601271b2c80a013bb0a82f465dc34cf5da7a875/crates/oxc_ast/src/ast_impl/js.rs#L898-L901 work?

Reply:
Resolved in:  32f319e8693ba9ce943c057216966fc19bc74dc6...81780ca214d2835e182ecf2257ab99b09fd94587